EIPS/eip-7480.md

```diff
@@ -0,0 +1,92 @@
+---
+eip: 7480
+title: EOF - Data section access instructions
+description: Instructions to read data section of EOF container
+author: Andrei Maiboroda (@gumb0), Alex Beregszaszi (@axic), Paweł Bylica (@chfast)
+discussions-to: https://ethereum-magicians.org/t/eip-7480-eof-data-instructions/15414
+status: Draft
+type: Standards Track
+category: Core
+created: 2023-08-11
+requires: 3540, 3670
+---
+
+## Abstract
+
+Four new instrutions are introduced, that allow to read EOF container's data section: `DATALOAD` loads 32-byte word to stack, `DATALOADN` loads 32-byte word to stack where the word is addressed by a static immediate argument, `DATASIZE` loads data section size and `DATACOPY` copies a segment of data section to memory. 
+
+## Motivation
+
+Clear separation between code and data is one of the main features of EOF1. Data section may contain anything, e.g. compiler's metadata, but to make it useful for smart contracts, EVM has to have instructions that allow to read from data section. Previously existing instructions for bytecode inspection (`CODECOPY`, `CODESIZE` etc.) are deprecated in EOF1 and cannot be used for this purpose.
+
+`DATALOAD`, `DATASIZE`, `DATACOPY` instructions combination follows the design of existing instructions for reading other kinds of data (return data, call data).
+
+`DATALOADN` is an optimized version of `DATALOAD`, where data offset to read is set at compilation time, and therefore need not be validated at run-time, which makes the instruction cheaper.
+
+## Specification
+
+We introduce four new instructions on the same block number [EIP-3540](./eip-3540.md) is activated on:
```

Comment:
i think it would be better to specify activation independently (and this EIP already specifies that it depends on 3540). Their activation is dependent on inclusion of the EIP in a particular hardfork which would be an independent decision by the ACD.

```suggestion
We introduce four new instructions:
```

Reply:
Most of EOF must be activated at the same time, otherwise the functionality is very limited.